packages/walletconnect/src/utils.ts

```diff
@@ -1,9 +1,24 @@
 /**
- * @param urls - An array of URLs to try to connect to.
+ * @param rpcMap - Map of chainIds to rpc url(s).
  * @param timeout - How long to wait before a call is considered failed, in ms.
  */
-export async function getBestUrl(urls: string[], timeout: number): Promise<string> {
+export async function getRpcBestUrlMap(rpcMap: Record<string, string | string[]>, timeout: number) {
+  return Object.fromEntries(
+    await Promise.all(
+      Object.entries(rpcMap).map(
+        async ([chainId, map]): Promise<[string, string]> => [`${chainId}`, await getBestUrl(map, timeout)]
```

Comment:
`chainId` is already a string

```suggestion
        async ([chainId, map]): Promise<[string, string]> => [chainId, await getBestUrl(map, timeout)]
```